server/controller/user-contoller.js

```diff
@@ -1,71 +1,86 @@
 const User = require("../model/User");
 const bcrypt = require("bcryptjs");
 
-const getAllUser = async(req,res,next) =>{
-    let users;
+const getAllUser = async (req, res, next) => {
+  let users;
 
-    try{
-        users = await User.find();
-    }
-    catch(err){
-        console.log(err);
-    }
-    if(!users){
-        return res.status(404).json({ message : "users are not found"})
-    }
+  try {
+    users = await User.find();
+  } catch (err) {
+    console.log(err);
+  }
+  if (!users) {
+    return res.status(404).json({ message: "users are not found" });
+  }
 
-    return res.status(200).json({users});
-}
+  return res.status(200).json({ users });
+};
 
-const signUp = async(req,res,next) =>{
-   const { name , email , password } = req.body;
+const signUp = async (req, res) => {
+  try {
+    const { name, password, email } = req.body;
 
-   let existingUser;
+    const existingUser = await User.findOne({ email });
+    if (!existingUser) {
+      const hashedPassword = bcrypt.hashSync(password);
+      const user = new User({
+        name,
+        email,
+        password: hashedPassword,
+        blogs: [],
+      });
+      await user.save();
+      return res.status(200).json({ message: "signup successfull" });
+    } else {
+      return res.status(200).json({ message: "user already exist" });
+    }
+  } catch (error) {
+    return res.status(500).json({ message: `${error}` });
+  }
+};
 
-   try{
-    existingUser = await User.findOne({email})
-   }catch(e){
-    console.log(err);
-   }
+// const logIn = async (req, res, next) => {
+//   const { email, password } = req.body;
 
-   if(existingUser){
-       return res.status(400).json({message : "User is already exists!"})
-   }
-   const hashedPassword = bcrypt.hashSync(password);
-   const user = new User({
-       name,email,
-       password: hashedPassword,
-       blogs: []
-   });
+//   let existingUser;
 
-   try{
-       user.save();
-       return res.status(201).json({ user })
-   }
-   catch(e){console.log(e);}
-}
+//   try {
+//     existingUser = await User.findOne({ email });
+//   } catch (err) {
+//     console.log(err);
+//   }
+//   if (!existingUser) {
+//     return res.status(404).json({ message: "User is not found" });
+//   }
```

Comment:
Best to not have dead code committed in the repository.

Reply:
Sure